### PR TITLE
Merge #132 (v1.15.0) in master

### PR DESCRIPTION
**Pull request by lruozzi9**
Merge #132 (v1.15.0) in master.

### CODE REVIEW

**Comment by lruozzi9** [comment]
Gentle reminder 😄.

**Comment by mmenozzi** [comment]
> Gentle reminder 😄.

What's the purpose of this PR? And why roave BC check is red?

**Comment by lruozzi9** [comment]
> What's the purpose of this PR? And why roave BC check is red?

That's the merge of PR #132 into master. Remember that the master branch actually contains the future v2.0 release. It already has the Akeneo official PIM client so this PR will adjust the feature of the #132 with that client. The BC is red because v2.0 will contain several BC breaks.

File: src/ValueHandler/ProductOptionValueHandler.php [hunk 1]
```diff
@@ -16,20 +18,24 @@
...
+
     public function __construct(
         private AkeneoPimClientInterface $apiClient,
         private ProductOptionRepositoryInterface $productOptionRepository,
         private FactoryInterface $productOptionValueFactory,
         private FactoryInterface $productOptionValueTranslationFactory,
         private RepositoryInterface $productOptionValueRepository,
-        TranslationLocaleProviderInterface $translationLocaleProvider = null
+        TranslationLocaleProviderInterface $translationLocaleProvider = null,
+        TranslatorInterface $translator = null
```
Comment (mmenozzi) [inline src/ValueHandler/ProductOptionValueHandler.php:38]:
```suggestion
        TranslatorInterface $translator,
        TranslationLocaleProviderInterface $translationLocaleProvider = null
```

BC break is allowed on master.

File: src/ValueHandler/ProductOptionValueHandler.php [hunk 2]
```diff
@@ -41,6 +47,16 @@ public function __construct(
...
         $this->translationLocaleProvider = $translationLocaleProvider;
+        if ($translator === null) {
+            trigger_deprecation(
+                'webgriffe/sylius-akeneo-plugin',
+                '1.15',
+                'Not passing a translator to %s is deprecated and will not be possible anymore in %s',
+                __CLASS__,
+                '2.0'
+            );
+        }
```
Comment (mmenozzi) [inline src/ValueHandler/ProductOptionValueHandler.php:58]:
```suggestion
```

BC break is allowed on master.

File: src/ValueHandler/ProductOptionValueHandler.php [hunk 3]
```diff
@@ -95,13 +161,85 @@ public function handle($productVariant, string $optionCode, array $akeneoValue):
...
+        $optionValue = $this->getOrCreateProductOptionValue($optionValueCode, $productOption);
+
+        /** @var string[] $locales */
+        $locales = $this->getLocaleCodes($product);
+
+        foreach ($locales as $localeCode) {
+            $label = $floatAmount . ' ' . $unit;
+            if ($this->translator !== null) {
+                $label = $this->translator->trans('webgriffe_sylius_akeneo.ui.metric_amount_unit', ['unit' => $unit, 'amount' => $floatAmount], null, $localeCode);
+            }
```
Comment (mmenozzi) [inline src/ValueHandler/ProductOptionValueHandler.php:213]:
```suggestion
            $label = $this->translator->trans('webgriffe_sylius_akeneo.ui.metric_amount_unit', ['unit' => $unit, 'amount' => $floatAmount], null, $localeCode);
```

File: src/ValueHandler/ProductOptionValueHandler.php [hunk 4]
```diff
@@ -95,13 +161,85 @@ public function handle($productVariant, string $optionCode, array $akeneoValue):
...
+
+        $optionValue = $this->getOrCreateProductOptionValue($optionValueCode, $productOption);
+
+        /** @var string[] $locales */
+        $locales = $this->getLocaleCodes($product);
+        foreach ($locales as $localeCode) {
+            $label = (string) $akeneoDataValue;
+            if ($this->translator !== null) {
+                $label = $akeneoDataValue ? $this->translator->trans('sylius.ui.yes_label', [], null, $localeCode) : $this->translator->trans('sylius.ui.no_label', [], null, $localeCode);
+            }
```
Comment (mmenozzi) [inline src/ValueHandler/ProductOptionValueHandler.php:233]:
```suggestion
            $label = $akeneoDataValue ? $this->translator->trans('sylius.ui.yes_label', [], null, $localeCode) : $this->translator->trans('sylius.ui.no_label', [], null, $localeCode);
```

**Review by mmenozzi — CHANGES_REQUESTED** [review]

File: src/ValueHandler/ProductOptionValueHandler.php (hunk 1 above)
Comment (lruozzi9) [reply src/ValueHandler/ProductOptionValueHandler.php:38]: Yes, but I will keep this type of changes on the #130.

File: src/ValueHandler/ProductOptionValueHandler.php (hunk 2 above)
Comment (lruozzi9) [reply src/ValueHandler/ProductOptionValueHandler.php:58]: Yes, but I will keep this kind of changes on the #130.

File: src/ValueHandler/ProductOptionValueHandler.php (hunk 3 above)
Comment (lruozzi9) [reply src/ValueHandler/ProductOptionValueHandler.php:213]: Yes, but I will keep this kind of changes on the #130.

File: src/ValueHandler/ProductOptionValueHandler.php (hunk 1 above)
Comment (mmenozzi) [reply src/ValueHandler/ProductOptionValueHandler.php:38]: Ok!

File: src/ValueHandler/ProductOptionValueHandler.php (hunk 2 above)
Comment (mmenozzi) [reply src/ValueHandler/ProductOptionValueHandler.php:58]: Ok!

File: src/ValueHandler/ProductOptionValueHandler.php (hunk 3 above)
Comment (mmenozzi) [reply src/ValueHandler/ProductOptionValueHandler.php:213]: Ok!

File: src/ValueHandler/ProductOptionValueHandler.php (hunk 4 above)
Comment (mmenozzi) [reply src/ValueHandler/ProductOptionValueHandler.php:233]: Ok!

**Review by mmenozzi — APPROVED** [review]

**Pull request merged**